src/main/kotlin/racingcar/domain/Car.kt

```diff
@@ -5,24 +5,31 @@ import racingcar.domain.movestrategy.RandomMoveStrategy
 
 class Car(
     val name: String,
-    position: Int = 0,
+    private val positionHistory: PositionHistory = PositionHistory.initPositionHistory(),
     private val moveStrategy: MoveStrategy = RandomMoveStrategy(),
 ) : Comparable<Car> {
 
     init {
-        require(name.length <= 5) { "자동차 이름은 5자를 초과할 수 없습니다." }
+        require(name.length <= MAX_NAME_LENGTH) { "자동차 이름은 5자를 초과할 수 없습니다." }
     }
 
-    var position = position
-        private set
-
     fun move() {
-        if (moveStrategy.canMove()) {
-            position += 1
-        }
+        positionHistory.add(moveStrategy.canMove())
+    }
+
+    fun getPositionOf(round: Int): Position {
+        return positionHistory.getPositionOf(round)
+    }
+
+    fun getCurrentPosition(): Position {
+        return positionHistory.getCurrentPosition()
     }
 
     override fun compareTo(other: Car): Int {
-        return position.compareTo(other.position)
+        return getCurrentPosition().compareTo(other.getCurrentPosition())
```

Comment:
compareTo 함수가 infix 함수이기에 다음과 같이 표현할수도 있습니다 
```suggestion
        return getCurrentPosition() compareTo other.getCurrentPosition()
```